chainerx_cc/chainerx/routines/math.cc

```diff
@@ -984,6 +984,41 @@ Array Arctan(const Array& x) {
     return out;
 }
 
+Array Arctan2(const Array& x1, const Array& x2) {
+    Array x1_cast = x1.dtype() < x2.dtype() ? x1.AsType(x2.dtype()) : x1;
+    Array x2_cast = x2.dtype() < x1.dtype() ? x2.AsType(x1.dtype()) : x2;
```

Comment:
The result dtype should be `GetMathResultDtype(GetArithmeticResultDtype(x1, x2))`

Reply:
Done.